### DISCUSSION TITLE
Trying to run `bash tools/init --no-gh`, failing

### DISCUSSION THREAD

**Discussion by blaurascon** [cotes2020/jekyll-theme-chirpy#1757] [Q&A] [answered]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### How did you create the site?

Built from `jekyll-theme-chirpy`

### Description

Trying to build from `jekyll-theme-chirpy` because I want to be able to change some of the CSS and that does not seem to be available on `chirpy-starter`. (If it is, please let me know - I was able to get that version working)

Trying to follow the initial setup step of running `bash tools/init --no-gh` (no-gh arg because I do not want to use Github Pages; I have another host). However, I'm on Windows. I don't have access to a Linux machine, nor am I able to dual boot (windows destroyed my last dual-boot partition). I have tried four separate ways of running the init file (below), but the same errors get thrown:

```tools/init: line 4: $'\r': command not found
: invalid option 5: set: -
set: usage: set [-abefhkmnptuvxBCHP] [-o option-name] [--] [arg ...]
tools/init: line 6: $'\r': command not found
tools/init: line 9: $'\r': command not found
tools/init: line 11: $'\r': command not found
tools/init: line 13: $'\r': command not found
tools/init: line 16: $'\r': command not found
tools/init: line 18: $'\r': command not found
tools/init: line 19: syntax error near unexpected token `$'{\r''
'ools/init: line 19: `help() {
```

The file looks like it cloned to my machine just fine.

I know I'm almost certainly missing something but I'm not sure where to check.

### Operations you have already tried

Running `bash tools/init --no-gh` through Command Line
Running `bash tools/init --no-gh` through Powershell
Running `bash tools/init --no-gh` through Windows Subsystem for Linux (Ubuntu)
Running `bash tools/init --no-gh` through Node.js Command Line

Running `bash tools/init` in the above, without the `--no-gh` arg

**Comment by NichtsHsu** [comment, marked as answer]
Windows uses a pair of CR and LF (i.e. `\n\r`) to terminate lines while UNIX uses an single LF (i.e. `\n`). When you clone repositories on Windows, git will replace all `\n` to `\n\r`, which makes bash unable to recognize `\r`.

You might try the following solutions:

1. Completely use the WSL, re-clone this repository in WSL and run bash.
2. [Configure `core.autocrlf` of git to avoid automatic conversion](https://stackoverflow.com/questions/2333424/distributing-git-configuration-with-the-code/2354278#2354278).

**Reply by blaurascon** [reply]
Oh that's fascinating, I had no idea. 

I went through WSL and re-cloned the repository, everything seems to be working properly now. Thank you!
